src/imitation/data/fragments.py

```diff
@@ -0,0 +1,87 @@
+import random
+import warnings
+from typing import Callable, List, Optional, Tuple
+
+from imitation.data.types import TrajectoryWithRew
+
+Fragmenter = Callable[
+    [List[TrajectoryWithRew]], List[Tuple[TrajectoryWithRew, TrajectoryWithRew]]
+]
+"""Creates pairs of trajectory fragments from a collection of trajectories."""
+
+
+class RandomFragmenter:
+    """Sample fragments of trajectories uniformly at random with replacement.
+
+    Note that each fragment is part of a single episode and has a fixed
+    length. This leads to a bias: transitions at the beginning and at the
+    end of episodes are less likely to occur as part of fragments (this affects
+    the first and last fragment_length transitions).
+
+    TODO(ejnnr): should we correct for this bias and make all transitions equally
+    likely to be part of fragments?
+    """
+
+    def __init__(
+        self,
+        fragment_length: int = 50,
+        num_pairs: int = 50,
+        seed: Optional[float] = None,
+    ):
+        """Initialize the fragmenter.
+
+        Args:
+            fragment_length: the length of each sampled fragment
+            num_pairs: the number of fragment pairs to sample
+            seed: an optional seed for the internal RNG
+        """
+        self.fragment_length = fragment_length
+        self.num_pairs = num_pairs
+        self.rng = random.Random(seed)
+
+    def __call__(
+        self, trajectories: List[TrajectoryWithRew]
+    ) -> List[Tuple[TrajectoryWithRew, TrajectoryWithRew]]:
+        fragments: List[TrajectoryWithRew] = []
+
+        # filter out all trajectories that are too short
+        trajectories = [
```

Comment:
This is also a bias, albeit an unavoidable one, but worth mentioning in docstring.

Also tempted to log how many trajectories are removed (just compute len before and after).